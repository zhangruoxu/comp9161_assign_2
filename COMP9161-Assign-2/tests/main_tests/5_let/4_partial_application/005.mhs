main
     =
    let f  =
        letfun f  x =
            letfun g  y = x * -y;

    in let z  = 
        letfun z  x = f 3;   -- discards 2, returns \y -> 3 * -y

    in z 2 4;
            
