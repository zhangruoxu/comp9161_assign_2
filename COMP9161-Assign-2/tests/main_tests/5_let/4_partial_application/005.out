main :: Int = let f :: (Int -> Int -> Int) = (letfun f :: (Int -> Int -> Int) x = (letfun g :: (Int -> Int) y = (*) x ((- y))));
              in let z :: forall h. (h -> Int -> Int) = (letfun z :: (h -> Int -> Int) x = f 3);
                 in z 2 4;
